### PR TITLE
remove AMENT_IGNORE check in clang-tidy when looking for compilation db

### PR DESCRIPTION
**Pull request by alsora**
The clang-tidy linter is currently broken in ROS 2 Rolling due to this PR https://github.com/ament/ament_cmake/pull/410

That PR adds an `AMENT_IGNORE` file in all packages build folders.
These folders is where the compilation database is also generated.
Clang-tidy needs the compilation database in order to run, but it currently skips directories with an `AMENT_IGNORE` file in them.

This PR removes this check which seems redundant.

### CODE REVIEW

**Review by clalancette — CHANGES_REQUESTED** [review]
After looking at this, we think the change is OK but we could definitely use a comment here explaining the situation (as laid out in https://github.com/ament/ament_cmake/pull/410#issuecomment-1588296267).

**Comment by alsora** [comment]
Added comment; sorry for the long wait, but this PR completely fell of my radar.

File: ament_clang_tidy/ament_clang_tidy/main.py [hunk 1]
```diff
@@ -277,9 +277,12 @@ def get_compilation_db_files(paths):
     for path in paths:
         if os.path.isdir(path):
             for dirpath, dirnames, filenames in os.walk(path):
-                if 'AMENT_IGNORE' in dirnames + filenames:
-                    dirnames[:] = []
-                    continue
+                # NOTE: here we don't check for the AMENT_IGNORE file.
+                # This function tries to find compile_commands.json file in the build folders.
+                # Build folders always include a AMENT_IGNORE file, so checking for it would result in not finding any compilation db.
+                # The check would also be redundant, because if a source folder was marked with AMENT_IGNORE, then
```
Comment (clalancette) [inline ament_clang_tidy/ament_clang_tidy/main.py:283]: Actually, flake8 thinks these lines are too long (they need to be < 100).  Once that is fixed we can run full CI on it.

**Review by clalancette — CHANGES_REQUESTED** [review]

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19730/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14236/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20456/)

**Pull request merged**